File: src/Other/PlusOne.java
package Other;

public class PlusOne {
    public int[] plusOne(int[] digits) {
        for (int i = digits.length - 1; i >=0; i--) {
            if (digits[i] != 9) {
                digits[i]++;
                break;
            } else {
                digits[i] = 0;
            }
        }
        if (digits[0] == 0) {
            int[] res = new int[digits.length+1];
            res[0] = 1;
            return res;
        }
        return digits;
    }

    public static void main(String args[]) {
        PlusOne p = new PlusOne();
        int[] input = {9,8,9,9};
        System.out.println("Plus one: " + p.plusOne(input));
    }
}

File: py/essential/PlusOne.py
# https://leetcode.com/problems/plus-one/?envType=problem-list-v2&envId=rr2ss0g5

from typing import List


class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        carry = 1
        n = len(digits)
        for i in range(n - 1, -1, -1):
            curSum = digits[i] + carry
            if curSum == 10:
                carry = 1
                curSum = 0
            else:
                carry = 0
            digits[i] = curSum
        if carry == 1:
            digits.insert(0, 1)
        return digits


s = Solution()
print(s.plusOne([4, 3, 2, 1]))
print(s.plusOne([9, 9]))
print(s.plusOne([7, 9, 9]))
